src/microvm/src/vmm/microvm/kvm/vmem.rs

```diff
@@ -45,14 +52,43 @@ pub struct VirtualMemory {
     /// Kernel location and size.
     kernel: Option<(u64, usize)>,
     /// Initial RAM disk location and size.
-    _initrd: Option<(u64, usize)>,
+    initrd: Option<(u64, usize)>,
     /// Control register used to inform the guest about the number of messages ready to be consumed.
     credits: u32,
 }
 
+///
+/// # Description
+///
+/// A structure that represents the header in virtual memory snapshot files.
+///
+#[repr(C)]
+struct SnapshotHeader {
+    /// Memory size (8 bytes): usize
+    memory_size: usize,
+    /// Kernel base (8 bytes): u64
+    kernel_base: u64,
+    /// Kernel size (8 bytes): usize
+    kernel_size: usize,
+    /// Initrd base (8 bytes): u64
+    initrd_base: u64,
+    /// Initrd size (8 bytes): usize
+    initrd_size: usize,
+    /// Credits (4 bytes): u32
+    credits: u32,
+    /// Padding (20 bytes): zeros
+    padding: [u8; 20],
```

Comment:
Create a private constant for this in `Self`. Also, document ther eason why this padding is required. If it is to make the structure 64-bytes long, consider adding a static check for that.